src/helpers/internal_viewport.ts

```diff
@@ -203,18 +203,34 @@ export class InternalViewport {
     this.adjustViewportZoneY();
   }
 
+  /**
+   *
+   * @param zone
+   * @returns Computes the absolute coordinate of a given zone inside the viewport
+   */
   getRect(zone: Zone): Rect | undefined {
     const targetZone = intersection(zone, this.zone);
     if (targetZone) {
+      const x =
+        this.getters.getColRowOffset("COL", this.zone.left, targetZone.left) +
+        this.offsetCorrectionX;
+
+      const y =
+        this.getters.getColRowOffset("ROW", this.zone.top, targetZone.top) + this.offsetCorrectionY;
+
+      const width = Math.min(
+        this.getters.getColRowOffset("COL", targetZone.left, targetZone.right + 1),
+        this.width
+      );
+      const height = Math.min(
+        this.getters.getColRowOffset("ROW", targetZone.top, targetZone.bottom + 1),
+        this.height
+      );
```

Comment:
Any chance you know what the "+1" are for ? Maybe add a comment to explain if you can

Reply:
the +1 have always been here, its because the boundary considers the start of a header and not the end. Don't think it deserves a specific comment. Same goes when you want to generate a range in python for example

Reply:
Oh yeah it's a col/row position, not pixels. My bad